lib/puppet/ssl/host.rb

```diff
@@ -559,13 +559,15 @@ def download_and_save_crl_bundle(store=nil)
       # If no SSL store was suppoed, use this host's SSL store
       store ||= ssl_store
       client = http_client(Puppet::Rest::SSLContext.new(OpenSSL::SSL::VERIFY_PEER, store))
+      crl_last_modified = Puppet::FileSystem.exist?(crl_path) ? File.mtime(crl_path) : nil
       Puppet::Util.replace_file(crl_path, 0644) do |file|
-        Puppet::Rest::Routes.get_crls(client, CA_NAME) do |chunk|
+        Puppet::Rest::Routes.get_crls(client, CA_NAME, crl_last_modified) do |chunk|
           file.write(chunk)
         end
       end
     rescue Puppet::Rest::ResponseError => e
-      raise Puppet::Error, _('Could not download CRLs: %{message}') % { message: e.message }
+      # if the ca_crl on master is not newer than the agent's, 304 will be returned and that's completely valid.
+      raise Puppet::Error, _('Could not download CRLs: %{message}') % { message: e.message } unless e.response.status_code == 304
```

Comment:
We also try to avoid postconditions unless the action is trivially long (some folks avoid them altogether).

We've found that when folks are scanning the code they often won't process the conditional when it's formatted like this.

Reply:
Done!